org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/batch/ClasspathJsr199.java

```diff
@@ -269,6 +329,11 @@ public IModule getModule(char[] name) {
 		return super.getModule(name);
 	}
 
+	@Override
+	public IModule getModule() {
+		return this.module;
```

Comment:
Called by the compiler. If we handle multiple modules, how should the compiler ask for a specific module?

At this point I wonder if the ClasspathJsr199 with Location MODULE_PATH should actually be resolved to a **set** of classpaths, just like ClasspathContainers in the IDE?

Reply:
Yep, looks like it's not adequate for such scenarios. ClasspathJsr199 is more of a version of ClasspathLocation that knows a bit about Location but not doing a good job of morphing into a Location.

Reply:
Should I try to move the loops from `ClasspathJsr199.initialize()` into `EclipseCompilerImpl.handleLocations()` so that we can create individual `ClasspathJsr199` per module?

Reply:
Done